naga/src/compact/mod.rs

```diff
@@ -264,6 +264,65 @@ pub fn compact(module: &mut crate::Module) {
             module_map.global_expressions.adjust(init);
         }
     }
+    // Adjust comments
+    if let Some(ref mut comments) = module.comments {
+        log::trace!("adjusting comments");
+        let crate::Comments {
+            module: _,
+            types: ref mut comment_types,
+            struct_members: ref mut comment_struct_members,
+            entry_points: _,
+            functions: _,
+            constants: ref mut comment_constants,
+            global_variables: _,
+        } = **comments;
+        log::trace!("adjusting comments for types");
+        for comment_type_handle in comment_types.keys().cloned().collect::<Vec<_>>() {
```

Comment:
I think in general we should be able to `std::mem::take` the `IndexMap`, and then do an owning iteration (just 'for (key, value) in map { ... }`) and check, adjust, and reinsert the values. This should be simpler than `swap_remove`.

Reply:
Thanks! I adressed this in 684e7e46c